src/main/java/br/com/sbs/cubatech/course/CourseController.java

```diff
@@ -39,9 +45,59 @@ public String getAll(@PathVariable String subCategoryCode,
     }
 
     @GetMapping("/admin/courses/new")
-    public String formNewCourse(){
+    public String formNewCourse(Model model){
+        List<SubCategory> subCategories = subCategoryRepository.findAll();
+        model.addAttribute("subCategories", subCategories);
+        model.addAttribute("courseVisibilityValues",CourseVisibility.values());
         return "/course/formNewCourse";
     }
 
+    @PostMapping("/admin/courses")
+    public String newCourse(@Valid CourseForm courseForm, BindingResult bindingResult, Model model){
+        if(bindingResult.hasErrors()){
+            return formNewCourse(model);
+        }
+        SubCategory subCategory = subCategoryRepository.findById(courseForm.getSubCategoryId())
+                .orElseThrow(()-> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+        Course course = courseForm.toEntity(subCategory);
+        courseRepository.save(course);
+        String categoryCode = subCategory.getCategory().getUrlCode();
```

Comment:
Sugestão usando a [Lei de Demeter](https://pt.wikipedia.org/wiki/Lei_de_Demeter):
Criar na `SubCategory` o método `getCategoryUrlCode()`
```suggestion
        String categoryCode = subCategory.getCategoryUrlCode();
```